robyn/dependencies.py

```diff
@@ -0,0 +1,38 @@
+
+from typing import Any, Callable, List, Dict
+from inspect import signature, Signature, Parameter
+from .utils import decode_bytearray
+import json
+
+def get_signature(call: Callable[..., Any]) -> Signature:
+    """
+    Gets the dependencies of the function wrapped by the routing decorators
+    (stores the name of the )
+    """
+    sign = signature(call)
+    params = [Parameter(
+        name = param.name,
+        kind = param.kind,
+        default = param.default,
+        # Add ForwardRef analysis here as the annotation value
```

Comment:
What is a ForwardRef? I am not sure if I understand it completely.

Reply:
This explains it perfectly: https://peps.python.org/pep-0484/#forward-references
Basically, type hints that use classes defined in other files cannot be resolved immediately because of how (circular) imports work in Python (although I don't know the specifics of that). Therefore, ForwardRefs are used when these annotations cannot be immediately resolved. I saw these being used when I was reverse engineering FastAPI's dependency injection system here: https://github.com/tiangolo/fastapi/blob/5905c3f740c8590f1a370e36b99b760f1ee7b828/fastapi/dependencies/utils.py#L248 in the get_typed_annotation function.
Since the callable's signature is retrieved and checked in a different file (inevitably) than it was defined in, any classes defined in the annotation would "external" relative to the function that was retrieving their signature. Thus we need to use ForwardReferences in virtually all cases where we have custom defined classes in the type hint. I am planning on working on this in a follow up PR to this one. I'm also going to try and eliminate external dependencies (like Pydantic) from this mechanism/implementation.

Reply:
I might be wrong about needing to use ForwardRefs, though. I will investigate further.